### ISSUE TITLE
runtime: fatal error: checkdead: runnable g [1.13 backport]

### ISSUE THREAD

**Issue by dmitshur** [golang/go#40503]
In a Go release meeting, we noticed issue #40368 likely affects Go 1.13 as well. This needs to be confirmed, and if so, this is the tracking issue for #40368 to be considered for backport to the next 1.13 minor release.

Including a copy of the rationale to backport from the 1.14 tracking issue #40398:

> This crash could affect any program running with GOMAXPROCS=1.

/cc @prattmic @aclements @cagedmantis @toothrot

**Comment by prattmic** [comment]
I believe this should affect 1.13 as well. I'll see if I can reproduce and send a cherry-pick.

**Comment by dmitshur** [comment]
Thanks Michael!

**Comment by prattmic** [comment]
So I've been unable to reproduce this on 1.13, though I believe it is still affected.

In general, the racing startm call must come from sysmon, as most other calls come from M's considered "running" by checkdead and would thus not trigger the throw. In 1.14+, sysmon is partially responsible for handling timers, so the timer expiration startm is an easy place for this race to occur.

In 1.13, sysmon was not responsible for timers, thus that startm doesn't exist, but there are two others: one for netpoll, and one for forcegc. Ironically, the netpoll case [already handles this exact problem](https://cs.opensource.google/go/go/+/master:src/runtime/proc.go;l=4689-4698;drc=6b4dcf19fa493905689dedc27f2232b74c366057) from #6070, which I hadn't noticed before now. I may be able to clean those up in 1.16 that we have another approach to this.

The forcegc should be subject to this problem, but it only runs every 2 minutes, so it is extremely rare. I'll still send the backport patch for consideration for this case, though I'm not sure how important it is.

**Comment by prattmic** [comment]
(I've filed #40518 to clean this up for 1.16).

**Comment by aclements** [comment]
I agree with your assessment that the possibility of this bug in 1.13 seems so astronomically small that I'm not sure it's worth fixing, but I'll leave the final decision to the release team.

**Comment by toothrot** [comment]
Approving. Even though it's astronomically small, we try to keep both versions at the same level of support for fixes.
